test/tools/yulInterpreter/Interpreter.h

```diff
@@ -96,6 +96,13 @@ struct InterpreterState
 	void dumpTraceAndState(std::ostream& _out) const;
 };
 
+/// Contains a function definition and the required context to call it
+struct FunctionContext
+{
+	std::map<YulString, FunctionContext> m_functions;
```

Comment:
You probably mean `FunctionDefinition const*` here, right?

Reply:
Hmm no. If it would be `FunctionDefinition const*` then the calling function would be missing the context again to call another function

Reply:
What looks suspicious here is that you store `FunctionContext` by value. This should produce an infinite data structure for the following code:
```
{
function f() {}
function g() {}
}
```

Since functions that are callable from the current context are always in the current scope or a parent scope, it should be enough to maintain a stack of scopes (we already have that) and just provide the list of visible functions as an initial segment of that stack, ending with the scope the function has been defined in.